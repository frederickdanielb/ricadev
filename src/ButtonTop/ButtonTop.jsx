import { faAngleUp } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { useEffect, useState } from "react";
const ButtonTop = ({ ...props }) => {
  const [isVisible, setIsVisible] = useState(false);

  const toggleVisibility = () => {
    if (window.pageYOffset > 300) {
      setIsVisible(true);
    } else {
      setIsVisible(false);
    }
  };

  const scrollToTop = () => {
    window.scrollTo({
      top: 0,
      behavior: "smooth",
    });
  };

  useEffect(() => {
    document.addEventListener("scroll", function (e) {
      toggleVisibility();
    });
  });

  return (
    <>
      {isVisible && (
        <div className="tap-top top" {...props} onClick={scrollToTop}>
          <div>
            <FontAwesomeIcon icon={faAngleUp} />
          </div>
        </div>
      )}
    </>
  );
};

export default ButtonTop;
