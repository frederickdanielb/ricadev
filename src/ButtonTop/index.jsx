import ButtonTop from "./ButtonTop";
export default ButtonTop;