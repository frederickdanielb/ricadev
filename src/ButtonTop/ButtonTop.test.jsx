import { act, render } from "@testing-library/react";
import ButtonTop from "./ButtonTop";
describe("ButtonTop component", () => {
	let component = null;
	beforeEach(() => {
		act(() => {
			component = render(
				<div
					style={{
						height: "300vh",
					}}
				>
					<ButtonTop />
				</div>
			);
		});
	});
	it("should render conrrectly", async () => {
		expect(component).toBeTruthy();
		expect(component.asFragment()).toMatchSnapshot();
	});
});
