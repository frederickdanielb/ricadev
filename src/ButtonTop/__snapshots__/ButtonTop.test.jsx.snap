// Vitest Snapshot v1

exports[`ButtonTop component > should render conrrectly 1`] = `
<DocumentFragment>
  <div
    style="height: 300vh;"
  />
</DocumentFragment>
`;
